### PR TITLE
bump Docker / Kubernetes version

### PR DESCRIPTION
**Pull request by tillepille**
Updates to Kubernetes 1.15 and Docker 19.03
Since Core DNS is a default feature, make general feature gates available. 

Lastly, tweaked the bootstrap script to make it more resilient

### CODE REVIEW

**Comment by tillepille** [comment]
rebased

**Comment by solidnerd** [comment]
@tillepille  This needs a rebase now. If you need further help let me know.

**Comment by tillepille** [comment]
@solidnerd done

File: main.tf [hunk 1]
```diff
@@ -40,7 +40,7 @@ resource "hcloud_server" "master" {
   }
 
   provisioner "remote-exec" {
-    inline = ["CORE_DNS=${var.core_dns} bash /root/master.sh"]
+    inline = "FEATURE_GATES=${var.feature_gates} bash /root/master.sh"
```
Comment (solidnerd) [inline main.tf]:
```suggestion
    inline = ["FEATURE_GATES=${var.feature_gates} bash /root/master.sh"]
```

File: variables.tf [hunk 2]
```diff
@@ -38,15 +38,16 @@ variable "ssh_public_key" {
 }
 
 variable "docker_version" {
-  default = "18.06"
+  default = "19.03"
 }
 
 variable "kubernetes_version" {
-  default = "1.12.2"
+  default = "1.15.4"
```
Comment (solidnerd) [inline variables.tf]: We should use here latest kubernetes `1.15.15` version

```suggestion
  default = "1.15.5"
```

**Comment by tillepille** [comment]
Both done @solidnerd , 1.15.4 was latest when creating ;-)

**Pull request merged**